enhancements/network/routed-ingress-support-for-primary-udn-attached-vms-with-static-ips.md

```diff
@@ -0,0 +1,879 @@
+(already happens today), and will also identify when the pod network attachment
+has a MAC address configuration request (defined in the KubeVirt
+`VMI.Spec.Domain.Devices.Interfaces` attribute).
+
+It will then access the `IPPool` (in the
+[centralized IP management option](#centralized-ip-management-workflow)) **or**
+the
+[de-centralized IP management option](#de-centralized-ip-management-workflow))
+to extract which IP addresses are assigned to said MAC address.
+
+Finally, the `ipam-extensions` mutating webhook will mutate the launcher pod to
+customize the primary UDN attachment using the multus default network
+annotation. This annotation (with an associated example) would look like:
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: pod-example
+  annotations:
+    v1.multus-cni.io/default-network: '{
+      "name": "isolated-net",
+      "namespace": "myisolatedns",
+      "mac": "02:03:04:05:06:07",
+      "ipam-claim-reference": "myvm.isolated-net",
+      "ips": [
+        "192.0.2.20/24",
+        "fd90:1234::14/64"
+      ]
+}'
+```
+
+OVN-Kubernetes will then act upon this information, by configuring the
+requested MAC and IPs in the pod. If the allocation of the IP is successful,
+said IPs will be persisted in the corresponding `IPAMClaim` CR (which already
+happens today). If it fails (e.g. that IP address is already in use in the
+subnet), the CNI will fail, crash-looping the pod. The error condition will be
+reported in the associated `IPAMClaim` CR, and an event logged in the pod.
+
+In the following sub-sections we will detail each of these changes.
+
+#### IPPool CRD
+This IPPool CRD has a 1:1 association to a UDN (or. For now, it'll only apply
+to a primary UDN though. In the future, nothing prevents these CRs from being
+used for secondary UDNs.
+
+The IPPool CRD is a cluster-scoped object associated to a UDN via the logical
+network name (`NAD.Spec.Config.Name` attribute), since we want to have this
+feature upstream in the k8snetworkplumbingwg, rather than in OVN-Kubernetes.
+
+The `IPPool` spec will have an attribute via which the admin can point to a
+cluster UDN - by the logical network name. The admin (which is the only actor
+able to create the `IPPool`) has read access to all NADs in all namespaces,
+hence they can inspect the NAD object to extract the network name. We could
+even update the cluster UDN type to feature the generated network name in its
+status sub-resource, to simplify the UX of the admin user.
+
+An alternative would be to reference the UDN by the NAD name - with that
+information, whatever controller reconciling the `IPPool` CRs can access the
+NAD, and extract the network name. This approach would require a validating
+webhook to ensure that you can't have multiple `IPPool`s referencing the same
+logical network (since multiple NADs with the same logical network name - in
+**different** namespaces can exist).
+
+### Preserving the VM gateway
```

Comment:
We have a huge issue with ipv6 here...

We are not able to override the OVN's RA with custom ipv6 gateway, since they are just the link local address generated from the gateway router mac address.

For pods is fine since ovn-k just tap into the network namespace and configure it with iplink.

Reply:
for ipv6 we'll just keep doing what we're doing today, i.e. ensure workloads only receive RAs with:
- LLA of the GW router of the node the workload is scheduled on
- lifetime = 0 (so it forgets the "old" GW)

Thus, I think the IPv6 flow is a bit different: when you migrate, we'll need to send a lifetime=0 RA with the address of the user provided GW configured in the NAD. 

Does it make sense ? I can put it into the enhancement.

Reply:
Done - added IPv6 considerations into the enhancement doc.